src/libraries/Pool.sol

```diff
@@ -384,7 +389,7 @@ library Pool {
             // if the protocol fee is on, calculate how much is owed, decrement feeAmount, and increment protocolFee
             if (cache.protocolFee > 0) {
                 // calculate the amount of the fee that should go to the protocol
-                uint256 delta = step.feeAmount * cache.protocolFee / ProtocolFeeLibrary.BIPS_DENOMINATOR;
+                uint256 delta = FullMath.mulDiv(step.feeAmount, cache.protocolFee, swapFee + cache.protocolFee);
```

Comment:
does this need mulDiv? I feel like it shouldnt be overflowing uint256

Reply:
`swapFee + cache.protocolFee` can be unchecked.